src/pages/api/identities/[...id].ts

```diff
@@ -169,6 +177,115 @@ const destroy: Handler = (request, response): Promise<void> => {
+    connectUnreachable: ERRORS_DATA.CONNECT_UNREACHABLE,
+    invalidBody: ERRORS_DATA.INVALID_BODY,
+    notFound: ERRORS_DATA.NOT_FOUND,
+  };
+
+  return getTracer().span("mark-identity-as-primary handler", async (span) => {
+    const userCookie = await getServerSideCookies<UserCookie>(request, {
+      cookieName: "user-cookie",
+      isCookieSealed: true,
+      cookieSalt: configVariables.cookieSalt,
+    });
+
+    if (!userCookie) {
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      response.end();
+      return;
+    }
+
+    const {
+      id: [identityId, requestedOperation],
+    } = request.query;
+
+    if (requestedOperation === "mark-as-primary") {
+      const isAuthorized = await isMarkingIdentityAsPrimaryAuthorized(
+        userCookie.sub,
+        identityId,
+      ).catch((error) => {
+        if (error instanceof GraphqlErrors) {
+          throw webErrorFactory({
+            ...webErrors.identityNotFound,
+            parentError: error,
+          });
+        }
+
+        if (error instanceof ConnectUnreachableError) {
+          throw webErrorFactory({
+            ...webErrors.connectUnreachable,
+            parentError: error,
+          });
+        }
+
+        throw error;
+      });
+
+      if (!isAuthorized) {
+        span.setDisclosedAttribute(
+          "is mark Identity as primary authorized",
+          false,
+        );
+
+        throw webErrorFactory(webErrors.badRequest);
+      }
+
+      span.setDisclosedAttribute(
+        "is mark Identity as primary authorized",
+        true,
+      );
+
+      return markIdentityAsPrimary(
+        configVariables.managementCredentials,
+        identityId,
+      )
+        .then((identity) => {
+          span.setDisclosedAttribute("is Identity marked as primary", true);
+
+          const alertMessage =
+            getIdentityType(identity.type) === IdentityTypes.EMAIL
+              ? `${identity.value} is now your primary email`
+              : `${identity.value} is now your primary phone number`;
+
+          setAlertMessagesCookie(response, [
+            generateAlertMessage(alertMessage),
+          ]);
+
+          response.statusCode = HttpStatus.OK;
+          response.setHeader("Content-type", "application/json");
+          response.end();
+          return;
+        })
+        .catch((error) => {
+          span.setDisclosedAttribute("is Identity marked as primary", false);
+
+          if (error instanceof GraphqlErrors) {
+            throw webErrorFactory({
+              ...webErrors.identityNotFound,
+              parentError: error,
+            });
+          }
+
+          if (error instanceof ConnectUnreachableError) {
+            throw webErrorFactory({
+              ...webErrors.connectUnreachable,
+              parentError: error,
+            });
+          }
+
+          throw error;
+        });
+    } else {
+      throw webErrorFactory(webErrors.notFound);
```

Comment:
if requestedOperation is in the url and not `=== "mark-as-primary"` it will throw a 404